src/components/Modals/SaveCourseModal.vue

```diff
@@ -38,30 +43,52 @@
 <script lang="ts">
 import { defineComponent } from 'vue';
 import TeleportModal from '@/components/Modals/TeleportModal.vue';
+import store from '@/store';
 
 export default defineComponent({
   components: { TeleportModal },
   props: {
     courseCode: { type: String, required: true },
-    collection: { type: String, default: 'No collections added yet' },
-    isdefaultCollection: { type: Boolean, default: true },
+  },
+  data() {
+    return {
+      checkedCollections: [] as string[], // New data property to manage checked state
+    };
+  },
+  computed: {
+    isDefaultCollection() {
+      const collections = store.state.collections.map(collection => collection.name);
+      return collections.length === 0;
+    },
+    collections() {
+      const collections = store.state.collections.map(collection => collection.name);
+      return collections.length === 0 ? ['No collections added yet'] : collections;
+    },
+    placeholder_name() {
+      const oldcollections = store.state.collections.map(collection => collection.name);
+      let newCollectionNum = 1;
```

Comment:
Could we store this in the state somehow? Iterating through `oldcollections` like this is probably discouraged by `eslint` for a reason. I know it goes counter to my advice from earlier, but in this case I feel like it would be reasonable to store the latest collection number.

Though that said, if this is just a placeholder route and a user would be forced to enter a name or something, then can leave as be (again, I appreciate that this is still a draft-level PR).

Reply:
I agree that storing this in state somehow would be better... also (for a very weird niche case that would be kind of annoying) if a renamed collection was named Collection 4 even if the collection was the second collection added, then the newCollectionNum wouldn't be updated correctly